autoscale/src/main/java/com/sequenceiq/periscope/init/InitializeMachineUserService.java

```diff
@@ -0,0 +1,43 @@
+package com.sequenceiq.periscope.init;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Strings;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.monitor.handler.CloudbreakCommunicator;
+import com.sequenceiq.periscope.service.AltusMachineUserService;
+import com.sequenceiq.periscope.service.ClusterService;
+
+@Component
+public class InitializeMachineUserService {
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private AltusMachineUserService altusMachineUserService;
+
+    @Inject
+    private CloudbreakCommunicator cloudbreakCommunicator;
+
+    @PostConstruct
+    public void init() {
+        List<Cluster> clusters = clusterService.findByEnvironmentCrnOrMachineUserCrn(null, null);
+        clusters.forEach(cluster ->  {
+            if (Strings.isNullOrEmpty(cluster.getEnvironmentCrn())) {
+                String envCrn = cloudbreakCommunicator.getAutoscaleClusterByCrn(cluster.getStackCrn()).getEnvironmentCrn();
```

Comment:
as dicussed, the API call in foreach can affect CB performance seriously, you should avoid it and for example use 1 list API call instead

Reply:
Updated this to a single API call.